Middleware/MiddlewarePipeline.cs

```diff
@@ -33,6 +35,54 @@ public class MiddlewarePipeline
             }
         }
 
+        private string ProcessSummonMessage(ISummonable middleware, IChatServiceConnection chatServiceConnection, ChatMessageModel message)
+        {
+            var injectVariables = new {
+                userId = chatServiceConnection.UserId,
+                userName = chatServiceConnection.UserName
+            };
+
+            // expect exactly 1 instance of the action verb
+            var verbMatchPattern = string.Concat(middleware.GetSummonVerb(), "{1}\\s?(");
+            // match for message format "<bot mention> <verb> *rest of message*" and return *rest of message* to middleware
+            var testPattern = string.Concat(
+                message.IsPrivateMessage ? chatServiceConnection.PrivateMessageRegExPrefix : chatServiceConnection.MentionMessageRegExPrefix, 
```

Comment:
Mentioned above but I think this pattern puts a lot of string manipulation burden on the Middleware